module/zfs/dnode.c

```diff
@@ -715,6 +712,8 @@ dnode_reallocate(dnode_t *dn, dmu_object_type_t ot, int blocksize,
 	}
 	rw_exit(&dn->dn_struct_rwlock);
 
+	VERIFY0(dnode_set_blksz(dn, blocksize, 0, tx));
```

Comment:
Is it necessary to set a new block size if the block size is not changing?

Reply:
dnode_set_blksz() has an early return for that.